devlib/trace/perf.py

```diff
@@ -108,7 +124,23 @@ def stop(self):
 
     # pylint: disable=arguments-differ
     def get_trace(self, outdir):
-        for label in self.labels:
+        for label, report_opt in zip(self.labels, self.report_optionstrings):
+            #in record mode generate report and copy
+            if self.perf_mode == 'record':
+                # .rpt
+                command = self._build_perf_report_command(report_opt, label)
+                self.target.execute(command, as_root=True)
+                target_file = self._get_target_reportfile(label)
+                host_relpath = os.path.basename(target_file)
+                host_file = _f(os.path.join(outdir, host_relpath))
+                self.target.pull(target_file, host_file)
+                # .trace
+                target_file = self._get_target_tracefile(label)
+                host_relpath = os.path.basename(target_file)
+                host_file = _f(os.path.join(outdir, host_relpath))
+                target_file = self._get_target_perfdatafile()
+                self.target.pull(target_file, host_file)
```

Comment:
The file never exists on the device because a mixture of perf behavior and how it is executed by WA. WA executes perf to run sleep and then kills it after a certain time. This seems to break the perf -o option and perf just outputs to a default place/file name ({current directory}/perf.data). So the -o option isn't used to run perf record and we do a rename of perf.data. Seemed easiest just to do it as part of the copy rather than directly on the device.

Reply:
> [..] then kills it after a certain time. This seems to break the perf -o option [...]

Interesting, I didn't know that. It's probably because `perf` has buffered some data and, as `.kill()` calls `kill` which sends `SIGKILL`, `perf` can't handle the signal and, therefore, can't write back what it's holding to the file. Sending a signal it's allowed to handle would probably solve that; this is something to keep in mind. Thanks